File: ts/example/src/tests/command_doubles.ts
import {ExampleCommand} from "../command";



export class ExampleCommandNull implements ExampleCommand {
  doSomething(foo: boolean, bar: number): void {}
}



export class ExampleCommandSpy implements ExampleCommand {
  // NOTE: Spies for Commands must have records of how the Command were called.
  calledArgs: [boolean, number][] = [];


  doSomething(foo: boolean, bar: number): void {
    // Records how were called.
    this.calledArgs.push([foo, bar]);
  }
}

File: swift/example/Tests/VanillaMockExampleTests/CommandDoubles.swift
@testable import VanillaMockExample


class ExampleCommandNull: ExampleCommandProtocol {
    func doSomething(foo: Bool, bar: Int) {}
}



class ExampleCommandSpy: ExampleCommandProtocol {
    enum CallArgs: Equatable {
        case doSomething(foo: Bool, bar: Int)
    }


    private(set) var callArgs = [CallArgs]()


    func doSomething(foo: Bool, bar: Int) {
        self.callArgs.append(.doSomething(foo: foo, bar: bar))
    }
}
